apps/full-site-editing/full-site-editing-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx

```diff
@@ -63,6 +62,57 @@ function WpcomBlockEditorNavSidebar() {
 		prevIsOpen.current = isOpen;
 	}, [ isOpen, prevIsOpen, setSidebarClosing ] );
 
+	const [ isListScrolling, setIsListScrolling ] = useState( false );
+
+	const observer = useRef< IntersectionObserver >();
+	const itemRefs = useRef< ( HTMLElement | null )[] >( [] );
+
+	const containerMount = ( el: HTMLDivElement | null ) => {
+		if ( el ) {
+			el.focus();
+		}
+
+		if ( observer.current ) {
+			observer.current.disconnect();
+		}
+
+		if ( ! window.IntersectionObserver ) {
+			return;
+		}
+
+		observer.current = new window.IntersectionObserver(
+			( entries ) => {
+				// If every iten is currently visible, then we're not scrolling
+				const notScrolling = entries.every( ( entry ) => entry.isIntersecting );
```

Comment:
this logic would read a bit simpler if it were inverted i.e. 
```
// If any item is not visible, then we're scrolling
const isScrolling = entries.some( ( entry ) => ! entry.isIntersecting );
setIsListScrrolling( isScrolling );
```

Reply:
Ya, notting nots is a bit weird sorry lol